### PR TITLE
vsr: fix reference count discipline for `primary_pipeline_prepare`

### PR DESCRIPTION
**Pull request by matklad**
By convention, functions accepting a Message "borrow" the argument, and functions returning a Message return an owned copy.

primary_pipeline_prepare and push_prepare violated this guideline.

Found this out during reconfig work, where `primary_pipeline_prepare` might actually drop the request if reconfig is in progress.

## Pre-merge checklist

Performance:

* [X] I am very sure this PR could not affect performance.

### CODE REVIEW

File: src/vsr/replica.zig [hunk 1]
```diff
@@ -4126,6 +4129,8 @@ pub fn ReplicaType(
             self.pipeline.queue.verify();
 
             const message = request.message;
+            defer self.message_bus.unref(message);
```
Comment (sentientwaffle) [inline src/vsr/replica.zig:4132]: Doesn't this go against "functions accepting a Message borrow the argument"?

Comment (matklad) [reply src/vsr/replica.zig:4132]: We accept a `Request` here, not the message. When call-site cosntructs a request, it does

```zig
            const request = .{
                .message = message.ref(),
                .realtime = realtime,
            };

            if (self.pipeline.queue.prepare_queue.full()) {
                self.pipeline.queue.push_request(request);
            } else {
                self.primary_pipeline_prepare(request);
            }
```

which makes sense --- every "stored" message is owned. 


But, yeah, I am not 100% happy here. But I am more happy with this than with 

```
pipeline_queue.push_prepare(prepare.ref());
```

Comment (sentientwaffle) [reply src/vsr/replica.zig:4132]: Ok I'm following now.

Before, we would "move" the `request.message` from the request queue to the prepare queue (`pipeline.queue.push_prepare`).
Now, `primary_pipeline_prepare` unrefs the message, and `pipeline.queue.push_prepare` takes its own reference.

**Review by sentientwaffle — APPROVED** [review]

File: src/vsr/replica.zig (hunk 1 above)
Comment (matklad) [reply src/vsr/replica.zig:4132]: >Before, we would "move" the request.message from the request queue to the prepare queue (pipeline.queue.push_prepare).

Yeah, and that was heavily dependent on there not being any stray `return`s on the path between the two.

**Pull request merged**